mp4/sdtp.go

```diff
@@ -0,0 +1,134 @@
+package mp4
+
+import (
+	"io"
+	"io/ioutil"
+)
+
+// SdtpBox - Sample Dependency Box (sdtp - optional)
+//
+// ISO/IEC 14496-12 Ed. 6 2020 Section 8.6.4
+// Contained in Sample Table Box (stbl)
+//
+// Table to determine whether a sample depends or is depended on by other samples
+type SdtpBox struct {
+	Version byte
+	Flags   uint32
+	Entries []SdtpEntry
+}
+
+// SdtpEntry (uint8)
+//
+// ISO/IEC 14496-12 Ed. 6 2020 Section 8.6.4.2
+type SdtpEntry uint8
+
+// IsLeading (bits 0-1)
+// 0: Leading unknown
+// 1: Has dependency before referenced I-picture (not decodable)
+// 2: Not a leading sample
+// 3: Has no dependency before referenced I-picture (decodable)
+func (entry SdtpEntry) IsLeading() uint8 {
+	return (uint8(entry) >> 6) & 3
+}
+
+// SampleDependsOn (bits 2-3)
+// 0: Dependency is unknown
+// 1: Depends on others (not an I-picture)
+// 2: Does not depend on others (I-picture)
+// 3: Reservced
+func (entry SdtpEntry) SampleDependsOn() uint8 {
+	return (uint8(entry) >> 4) & 3
+}
+
+// SampleIsDependedOn (bits 4-5)
+// 0: Dependency unknown
+// 1: Other samples may depend on this (not disposable)
+// 2: No other samples depend on this (disposable)
+// 3: Reserved
+func (entry SdtpEntry) SampleIsDependedOn() uint8 {
+	return (uint8(entry) >> 2) & 3
+}
+
+// SampleIsDependedOn (bits 6-7)
+// 0: Redundant coding unknown
+// 1: Redundant coding in this sample
+// 2: No redundant coding in this sample
+// 3: Reserved
+func (entry SdtpEntry) SampleHasRedundancy() uint8 {
+	return uint8(entry) & 3
+}
+
+// CreateSdtpBox - Create a new SdtpBox
+func CreateSdtpBox(entries []SdtpEntry) *SdtpBox {
+	return &SdtpBox{
+		Entries: entries,
+	}
+}
+
+// DecodeSdtp - box-specific decode
+func DecodeSdtp(hdr *boxHeader, startPos uint64, r io.Reader) (Box, error) {
+	data, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	s := NewSliceReader(data)
+	versionAndFlags := s.ReadUint32()
+	version := byte(versionAndFlags >> 24)
+	flags := versionAndFlags & flagsMask
+
+	// Supposed to get count from stsz
+	entries := make([]SdtpEntry, len(data)-s.pos)
```

Comment:
There is a method s.NrRemainingBytes() that can be used instead of len(data) - s.pos. I'll change that as I merge this PR.